core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableRel.java

```diff
@@ -27,7 +27,9 @@
  */
 public interface EnumerableRel
     extends RelNode {
-  RelFactories.FilterFactory FILTER_FACTORY = EnumerableFilter::create;
+  RelFactories.FilterFactory FILTER_FACTORY =
```

Comment:
Shouldn't we check that `correlVariables` is actually empty? Maybe add a new create `EnumerableFilter#create()` with the 3 args?

Reply:
Done